src/overlays/actors/ovl_En_Bigokuta/z_en_bigokuta.c

```diff
@@ -28,99 +43,640 @@ const ActorInit En_Bigokuta_InitVars = {
+}
+
+void EnBigokuta_SetupIdle(EnBigokuta* this) {
+    Animation_Change(&this->skelAnime, &gBigOctoIdleAnim, 0.5f, 0.0f, 0.0f, 1, -3.0f);
+    this->actionFunc = EnBigokuta_Idle;
+}
+
+void EnBigokuta_Idle(EnBigokuta* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    SkelAnime_Update(&this->skelAnime);
+    Math_StepToF(&this->actor.world.pos.y, this->actor.home.pos.y - 99.0f, 2.5f);
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 0x1000);
+    if ((this->actor.xzDistToPlayer < 300.0f) && ((player->actor.world.pos.y - this->actor.home.pos.y) < 100.0f)) {
+        EnBigokuta_SetupRise(this, globalCtx);
+    }
+}
+
+void EnBigokuta_SetupRise(EnBigokuta* this, GlobalContext* globalCtx) {
+    Vec3f splashPos;
+    s32 i;
+    s16 angle = 0;
+
+    Animation_PlayOnce(&this->skelAnime, &gBigOctoRiseOutOfWaterAnim);
+    splashPos.y = this->actor.home.pos.y;
+    for (i = 0; i < 8; i++) {
+        splashPos.x = Math_SinS(angle) * 70.0f + this->actor.world.pos.x;
+        splashPos.z = Math_CosS(angle) * 70.0f + this->actor.world.pos.z;
+        EffectSsGSplash_Spawn(globalCtx, &splashPos, NULL, NULL, 0, Rand_S16Offset(1000, 200));
+        angle = BINANG_ADD(angle, 0x2000);
+    }
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_DAIOCTA_LAND);
+    this->actionFunc = EnBigokuta_RiseOutOfWater;
+}
+
+void EnBigokuta_RiseOutOfWater(EnBigokuta* this, GlobalContext* globalCtx) {
+    Math_StepToF(&this->actor.world.pos.y, this->actor.home.pos.y - 16.5f, 15.0f);
+    if (SkelAnime_Update(&this->skelAnime)) {
+        EnBigokuta_SetupIdleAboveWater(this);
+    }
+}
+
+void EnBigokuta_SetupIdleAboveWater(EnBigokuta* this) {
+    Animation_MorphToLoop(&this->skelAnime, &gBigOctoIdleAnim, -5.0f);
+    this->actionFunc = EnBigokuta_IdleAboveWater;
+}
+
+void EnBigokuta_IdleAboveWater(EnBigokuta* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    Math_StepToF(&this->actor.world.pos.y, this->actor.home.pos.y - 16.5f, 2.5f);
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 0x1000);
+
+    if ((this->actor.xzDistToPlayer > 400.0f) || (this->actor.playerHeightRel > 200.0f)) {
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_DAIOCTA_SINK);
+        EnBigokuta_SetupIdle(this);
+    } else if ((this->actor.xzDistToPlayer < 200.0f) && globalCtx->grabPlayer(globalCtx, GET_PLAYER(globalCtx))) {
+        EnBigokuta_SetupSuckInPlayer(this, globalCtx);
+    }
+}
+
+void EnBigokuta_UpdateOrSetupCam(EnBigokuta* this, GlobalContext* globalCtx) {
+
+    if (this->actor.cutscene != -1) {
+        if (this->camId != 0) {
+            EnBigokuta_MoveCamera(this, globalCtx);
+        } else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+            Camera* camera = Play_GetCamera(globalCtx, 0);
+            EnBigokuta_SetupCutsceneCamera(this, globalCtx, &camera->at, &camera->eye);
+        } else {
```

Comment:
Add empty line between var declaration and the rest of the code
```suggestion
        } else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
            Camera* camera = Play_GetCamera(globalCtx, 0);

            EnBigokuta_SetupCutsceneCamera(this, globalCtx, &camera->at, &camera->eye);
        } else {
```